<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphHierarchiesEditorData</class>
 <widget class="QWidget" name="GraphHierarchiesEditorData">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>597</width>
    <height>417</height>
   </rect>
  </property>
  <layout class="QVBoxLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="HeaderFrame" name="header" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title" stdset="0">
      <string>Graphs</string>
     </property>
     <property name="expandable" stdset="0">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="treeFrame">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="CustomTreeView" name="hierarchiesTree">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::EditKeyPressed</set>
        </property>
        <property name="dragEnabled">
         <bool>true</bool>
        </property>
        <property name="dragDropMode">
         <enum>QAbstractItemView::DragOnly</enum>
        </property>
        <property name="defaultDropAction">
         <enum>Qt::CopyAction</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="indentation">
         <number>10</number>
        </property>
        <property name="uniformRowHeights">
         <bool>true</bool>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="allColumnsShowFocus">
         <bool>true</bool>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <attribute name="headerDefaultSectionSize">
         <number>50</number>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="bottomFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>27</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>27</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="bottomFrame" stdset="0">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="selectionLabel">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <action name="actionAdd_sub_graph">
   <property name="text">
    <string>Create empty subgraph</string>
   </property>
   <property name="toolTip">
    <string>Create an empty subgraph of the highlighted graph</string>
   </property>
  </action>
  <action name="actionClone_subgraph">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/22/hierarchy.png</normaloff>:/tulip/graphperspective/icons/22/hierarchy.png</iconset>
   </property>
   <property name="text">
    <string>Create clone subgraph</string>
   </property>
   <property name="toolTip">
    <string>Clone the highlighted graph into a subgraph</string>
   </property>
  </action>
  <action name="actionClone_sibling">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/22/sibling-clone.png</normaloff>:/tulip/graphperspective/icons/22/sibling-clone.png</iconset>
   </property>
   <property name="text">
    <string>Create clone sibling</string>
   </property>
   <property name="toolTip">
    <string>Clone the highlighted graph into a sibling of its parent graph</string>
   </property>
  </action>
  <action name="actionClone_sibling_with_properties">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/22/sibling-clone-props.png</normaloff>:/tulip/graphperspective/icons/22/sibling-clone-props.png</iconset>
   </property>
   <property name="text">
    <string>Create clone sibling (with properties)</string>
   </property>
   <property name="toolTip">
    <string>Clone the highlighted graph and its local properties into a sibling of its parent graph</string>
   </property>
  </action>
  <action name="actionDelete_graph">
   <property name="icon">
    <iconset resource="../../../../library/tulip-gui/resources/TulipGUIResource.qrc">
     <normaloff>:/tulip/gui/icons/i_del.png</normaloff>:/tulip/gui/icons/i_del.png</iconset>
   </property>
   <property name="text">
    <string>Delete graph</string>
   </property>
   <property name="toolTip">
    <string>Delete the highlighted graph, reparenting its subgraphs hierarchy to its parent</string>
   </property>
  </action>
  <action name="actionDelete_All">
   <property name="icon">
    <iconset resource="../../../../library/tulip-gui/resources/TulipGUIResource.qrc">
     <normaloff>:/tulip/gui/icons/i_del.png</normaloff>:/tulip/gui/icons/i_del.png</iconset>
   </property>
   <property name="text">
    <string>Delete all</string>
   </property>
   <property name="toolTip">
    <string>Delete the highlighted graph and all its subgraphs [Shift+Del]</string>
   </property>
   <property name="shortcut">
    <string>Shift+Del</string>
   </property>
  </action>
  <action name="actionRename">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/16/rename.png</normaloff>:/tulip/graphperspective/icons/16/rename.png</iconset>
   </property>
   <property name="text">
    <string>Rename</string>
   </property>
   <property name="toolTip">
    <string>Rename the highlighted graph</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionSave_to_file">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/16/document-save.png</normaloff>:/tulip/graphperspective/icons/16/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Export the highlighted graph into a file using the tlp/tlpb file format</string>
   </property>
  </action>
  <action name="actionCreate_induced_sub_graph">
   <property name="icon">
    <iconset resource="../resources/GraphPerspective.qrc">
     <normaloff>:/tulip/graphperspective/icons/22/hierarchy-select.png</normaloff>:/tulip/graphperspective/icons/22/hierarchy-select.png</iconset>
   </property>
   <property name="text">
    <string>Create subgraph from selection</string>
   </property>
   <property name="toolTip">
    <string>Create a subgraph from the selected elements of the highlighted graph [Ctrl+Shift+G]</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+G</string>
   </property>
  </action>
  <action name="actionExpand_hierarchy">
   <property name="text">
    <string>Expand all hierarchy</string>
   </property>
   <property name="toolTip">
    <string>Expand all the hierarchy under the current graph</string>
   </property>
  </action>
  <action name="actionCollapse_hierarchy">
   <property name="text">
    <string>Collapse all hierarchy</string>
   </property>
   <property name="toolTip">
    <string>Collapse all the hierarchy under the current graph</string>
   </property>
  </action>
  <action name="actionDelete_selection">
   <property name="text">
    <string>Delete selection</string>
   </property>
   <property name="toolTip">
    <string>Delete the currently selected nodes or edges</string>
   </property>
  </action>
  <action name="actionDelete_all_nodes">
   <property name="text">
    <string>Delete all nodes</string>
   </property>
   <property name="toolTip">
    <string>Delete all the nodes of the highlighted graph</string>
   </property>
  </action>
  <action name="actionDelete_all_edges">
   <property name="text">
    <string>Delete all edges</string>
   </property>
   <property name="toolTip">
    <string>Delete all the edges of the highlighted graph</string>
   </property>
  </action>
  <action name="actionDelete_selection_from_root_graph">
   <property name="text">
    <string>Delete selection from the root graph</string>
   </property>
   <property name="toolTip">
    <string>Delete selected elements from all the graph hierarchy</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HeaderFrame</class>
   <extends>QWidget</extends>
   <header>HeaderFrame.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CustomTreeView</class>
   <extends>QTreeView</extends>
   <header location="global">GraphHierarchiesEditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/GraphPerspective.qrc"/>
  <include location="../../../../library/tulip-gui/resources/TulipGUIResource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>hierarchiesTree</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>contextMenuRequested(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_sub_graph</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>addSubGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClone_subgraph</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>cloneSubGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClone_sibling</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>cloneSibling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClone_sibling_with_properties</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>cloneSiblingWithProperties()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_graph</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_All</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delAllGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExpand_hierarchy</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>expandGraphHierarchy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCollapse_hierarchy</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>collapseGraphHierarchy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRename</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>renameGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hierarchiesTree</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>doubleClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_to_file</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>saveGraphHierarchyInTlpFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCreate_induced_sub_graph</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>addInducedSubGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_all_edges</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delAllEdges()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_all_nodes</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delAllNodes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_selection</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delSelection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete_selection_from_root_graph</sender>
   <signal>triggered()</signal>
   <receiver>GraphHierarchiesEditorData</receiver>
   <slot>delSelectionFromRoot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>activeGraphSelected(QModelIndex)</slot>
  <slot>contextMenuRequested(QPoint)</slot>
  <slot>addSubGraph()</slot>
  <slot>delGraph()</slot>
  <slot>cloneSubGraph()</slot>
  <slot>cloneSibling()</slot>
  <slot>delAllGraph()</slot>
  <slot>delAllNodes()</slot>
  <slot>delAllEdges()</slot>
  <slot>delSelection()</slot>
  <slot>delSelectionFromRoot()</slot>
  <slot>createPanel()</slot>
  <slot>collapseGraphHierarchy()</slot>
  <slot>expandGraphHierarchy()</slot>
  <slot>renameGraph()</slot>
  <slot>doubleClicked(QModelIndex)</slot>
  <slot>saveGraphToFile()</slot>
  <slot>addInducedSubGraph()</slot>
 </slots>
</ui>
